services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -1778,7 +1802,8 @@ public Map<String, String> getFutureVersionsForMultiColos(String clusterName, St
     for (Map.Entry<String, ControllerClient> entry: controllerClients.entrySet()) {
       String region = entry.getKey();
       ControllerClient controllerClient = entry.getValue();
-      MultiStoreStatusResponse response = controllerClient.getFutureVersions(clusterName, storeName);
+      MultiStoreStatusResponse response =
+          controllerClient.retryableRequest(10, c -> c.getFutureVersions(clusterName, storeName));
       if (response.isError()) {
```

Comment:
Is this change relevant to the PR or is it adding more resiliency?

I wonder if this changes the behavior of the system in the case of transient issues where previously the function would return result with `IGNORED_CURRENT_VERSION` while the newer behavior upon retries can return actual results.

Do we need to be concerned with the behavior changes?

Reply:
We would rather get actual results, this change is just adding resiliency.